rasa/core/policies/ensemble.py

```diff
@@ -346,76 +346,193 @@ def get_state_featurizer_from_dict(cls, featurizer_config) -> Tuple[Any, Any]:
         return state_featurizer_func, state_featurizer_config
 
 
+class Prediction(NamedTuple):
+    """Stores the probabilities and the priority of the prediction."""
+
+    probabilities: List[float]
+    priority: int
+
+
 class SimplePolicyEnsemble(PolicyEnsemble):
     @staticmethod
-    def is_not_memo_policy(best_policy_name) -> bool:
-        is_memo = best_policy_name.endswith("_" + MemoizationPolicy.__name__)
-        is_augmented = best_policy_name.endswith(
-            "_" + AugmentedMemoizationPolicy.__name__
-        )
-        return not (is_memo or is_augmented)
+    def is_not_memo_policy(
+        policy_name: Text, max_confidence: Optional[float] = None
+    ) -> bool:
+        is_memo = policy_name.endswith("_" + MemoizationPolicy.__name__)
+        is_augmented = policy_name.endswith("_" + AugmentedMemoizationPolicy.__name__)
+        # also check if confidence is 0, than it cannot be count as prediction
+        return not (is_memo or is_augmented) or max_confidence == 0.0
 
-    def probabilities_using_best_policy(
-        self, tracker: DialogueStateTracker, domain: Domain
+    @staticmethod
+    def _is_not_mapping_policy(
+        policy_name: Text, max_confidence: Optional[float] = None
```

Comment:
Otherwise I'd also consider a `max_confidence==None`
```suggestion
        policy_name: Text, max_confidence: float = 0
```

Reply:
the problem is that `None` and `0` are different here, if it is `None` we don't check for it. E.g. `is_not_memo_` method is used somewhere else on policy without prediction

Reply:
we explicitly check if max_confidence is 0, because it means some policy set it to zero